src/bits/mod.rs

```diff
@@ -1,19 +1,16 @@
 use crate::secret_sharing::BooleanShare;
 use std::ops::{BitAnd, BitAndAssign, BitOr, BitOrAssign, BitXor, BitXorAssign, Index, Not};
 
-mod bit_array;
-
-pub use bit_array::BitArray64;
+pub mod bit_array;
 
 /// Trait for data types storing arbitrary number of bits.
 // TODO: Implement `Message`
-pub trait BitArray: BooleanShare + TryFrom<u128> + Index<usize> {
+pub trait BitArray: BooleanShare + Index<usize> {
     /// A bit array with all its elements initialized to 0.
     const ZERO: Self;
 
     /// Truncates the higher-order bits larger than `Self::BITS`, and converts
-    /// into this data type. This conversion is lossy. Callers are encouraged
-    /// to use `try_from` if the input is not known in advance.
+    /// into this data type. This conversion is lossy.
     fn truncate_from<T: Into<u128>>(v: T) -> Self;
```

Comment:
This could be marked `#[cfg(test)]` to avoid accidental misuse.

Reply:
This will be used in production. I spoke with Alex and this is the name we agreed on for the lossy conversion. Alternatively, we could use BitSlice or bitarr! macro to create a new BiArray instance, but neither is not user-friendly. I think we'll almost always use this function. Or we might need to bring back TryFrom<u128>.